### PR TITLE
Disable (long double, epsilon) numeric traits check with ibm/xl

### PR DESCRIPTION
**Pull request by ndellingwood**
Comparison check results in static_assert

### CODE REVIEW

**Comment by ndellingwood** [comment]
The failure is showing up in builds on Power9+Volta70 with the following configuration:

```
module load cmake/3.19.3 cuda/10.1.105 gcc/7.4.0 ibm/xl/16.1.1

cmake -DCMAKE_CXX_COMPILER=/home/projects/ppc64le/ibm/xlC/16.1.1/bin/xlC -DCMAKE_CXX_FLAGS="-qxflag=disable__cplusplusOverride -Wno-unused-function -O3 -Wall -Wunused-parameter -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_SERIAL=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ARCH_POWER9=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DKokkos_ENABLE_CUDA_LAMBDA=ON -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF $KOKKOS_PATH
```

**Comment by ndellingwood** [comment]
@dalg24 I wasn't sure how best to resolve without disabling the test for ibm/xl, I'm having issues connecting to slack to discuss (vpn?), but I will close if an alternative is preferred

File: core/unit_test/TestNumericTraits.hpp [hunk 1]
```diff
@@ -411,7 +411,9 @@ CHECK_SAME_AS_NUMERIC_LIMITS_MEMBER_FUNCTION(double, infinity);
 CHECK_SAME_AS_NUMERIC_LIMITS_MEMBER_FUNCTION(long double, infinity);
 CHECK_SAME_AS_NUMERIC_LIMITS_MEMBER_FUNCTION(float, epsilon);
 CHECK_SAME_AS_NUMERIC_LIMITS_MEMBER_FUNCTION(double, epsilon);
+#ifndef KOKKOS_COMPILER_IBM
```
Comment (dalg24) [inline core/unit_test/TestNumericTraits.hpp:414]:
```suggestion
#ifndef KOKKOS_COMPILER_IBM  // fails with XL 16.1.1
```

**Review by masterleinad — APPROVED** [review]
Fine with me.

**Comment by dalg24** [comment]
Were these failing since #4281 was merged a month ago or so?

**Comment by ndellingwood** [comment]
> Was this build failing since #4281 was merged a month ago or so?

@dalg24 Possibly, there was some downtime in this build due to migrating to a new system and some issues with xl config files and I had not monitored this build closely enough for a couple weeks. I can test a before/after merge of #4281 to confirm if you'd like

**Comment by dalg24** [comment]
No need to

**Review by dalg24 — APPROVED** [review]

**Pull request merged**